### PR TITLE
New syntax conversion for Array in boot.jl

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Review by JeffBezanson — APPROVED** [review]

File: base/boot.jl [hunk 1]
```diff
@@ -264,7 +264,7 @@ UnionAll(v::TypeVar, t::ANY) = ccall(:jl_type_unionall, Any, (Any, Any), v, t)
 
 Void() = nothing
 
-(::Type{Tuple{}})() = ()
+Tuple{}() where {} = ()
```
Comment (musm) [inline base/boot.jl]: Is this preferable, I guess so since the old may be deprecated?

Comment (martinholters) [inline base/boot.jl]: Does it even need the `where {}` part?

Comment (TotalVerb) [inline base/boot.jl]: Otherwise I imagine it gets interpreted as an empty static parameter list...

Comment (martinholters) [inline base/boot.jl]: Right, `Tuple{}() = ()` would be equal to `Tuple() = ()`.

File: base/boot.jl [hunk 2]
```diff
@@ -298,29 +298,28 @@ convert(::Type{T}, x::T) where {T} = x
...
-(::Type{Array{T,1}})(d::NTuple{1,Int}) where {T} = Array{T,1}(getfield(d,1))
-(::Type{Array{T,2}})(d::NTuple{2,Int}) where {T} = Array{T,2}(getfield(d,1), getfield(d,2))
-(::Type{Array{T,3}})(d::NTuple{3,Int}) where {T} = Array{T,3}(getfield(d,1), getfield(d,2), getfield(d,3))
-(::Type{Array{T,N}})(d::Vararg{Int, N}) where {T,N} = ccall(:jl_new_array, Array{T,N}, (Any,Any), Array{T,N}, d)
-(::Type{Array{T,1}})(m::Int) where {T} =
-    ccall(:jl_alloc_array_1d, Array{T,1}, (Any,Int), Array{T,1}, m)
-(::Type{Array{T,2}})(m::Int, n::Int) where {T} =
+Array{T,1}(d::NTuple{1,Int}) where {T} = Array{T,1}(getfield(d,1))
+Array{T,2}(d::NTuple{2,Int}) where {T} = Array{T,2}(getfield(d,1), getfield(d,2))
+Array{T,3}(d::NTuple{3,Int}) where {T} = Array{T,3}(getfield(d,1), getfield(d,2), getfield(d,3))
```
Comment (Sacha0) [inline base/boot.jl]: Perhaps add spaces separating function call arguments above and tuple elements below while you're at it? :)

**Review by Sacha0 — APPROVED** [review]
Thanks @musm!

File: base/boot.jl (hunk 1 above)
Comment (StefanKarpinski) [inline base/boot.jl]: I don't think this is the preferred form, but it's @JeffBezanson's call.

Comment (musm) [inline base/boot.jl]: After deprecation this should be  `Tuple{}() = ()` but now needs the `where {}`

Comment (JeffBezanson) [inline base/boot.jl]: I guess this doesn't really bother me much.

Comment (musm) [inline base/boot.jl]: I think I'll just keep this as it was

File: base/boot.jl (hunk 2 above)
Comment (Sacha0) [inline base/boot.jl]: (`getfield` above)

Comment (musm) [inline base/boot.jl]: I think that's fine for now

**Comment by musm** [comment]
unrelated time out on  travis

**Pull request merged**